README.md

```diff
@@ -4,47 +4,109 @@
 uftrace
 =======
 
-The uftrace tool is to trace and analyze execution of a program
-written in C/C++.  It was heavily inspired by the ftrace framework
-of the Linux kernel (especially function graph tracer) and supports
-userspace programs.  It supports various kind of commands and filters
-to help analysis of the program execution and performance.
+uftrace is a function call graph tracer for C, C++, Rust and Python programs.
+
+uftrace traces, records and analyzes the execution of programs written in
+C, C++ and Rust. It was heavily inspired by the ftrace framework of the
+Linux kernel (especially the function graph tracer) and supports userspace
+programs, and with elevated privileges, also kernel function call graphs.
+
+The name uftrace stems from the combination of user and ftrace:
+
+It can record data from:
+- User space functions by dynamically patching functions at runtime, or with code
+  compiled with with `-pg`, `-fpatchable-function-entry`, `-mfentry`, `-mnop-mcount`,
+  `-finstrument-functions`, or using clang with `-fxray-instrument`.
+- Library functions (through PLT hooking)
+- Kernel functions (using the ftrace framework in Linux kernel)
+- Kernel trace events affecting the program, e.g. scheduling events (which affect the
+  execution timing of the program) and records nanosecond-exact timestamps
+  (using perf_event and systemtap SDT)
+
+With the recorded data, it:
+- Can show colored and nested function call graphs
+- Can show arguments and return values symbolically using libc function prototypes
+  and DWARF debug information.
+- Can apply filters like minimal function call duration
+- Can extract metadata from traces (e.g. system information on which the trace was taken)
+- Can generate a symbol table of the traced program and library functions
+- When the execution of multiple programs was recorded,
+  it can generate a task relationship tree (parent and children) from the trace.
```

Comment:
It also can generate chrome trace data, flame graph or call-graph diagrams for graphviz and mermaid.

Reply:
Thanks, merged into the new paragraph about visualization from https://www.blackhat.com/asia-23/arsenal/schedule/#uftrace-dynamic-analysistracing-tool-for-ccrust-programs-30951:

> uftrace offers various filters to reduce the amount of trace data and provides
visualization using Chrome trace viewer and flame graph or call-graph diagrams
for graphviz and mermaid, allowing for a big picture view of the execution flow.